app/controllers/registrations_controller.rb

```diff
@@ -0,0 +1,59 @@
+class RegistrationsController < Devise::RegistrationsController
+
+  def new
+    super
```

Comment:
Maybe it will help (to create correct form on view):
```
    @user_locations = resource.user_locations.build
    @locations = @user_locations.build_location
```